R/target_market_share.R

```diff
@@ -470,6 +345,104 @@ aggregate_by_loan_id <- function(data) {
     ungroup()
 }
 
+format_output_dataframe <- function(data){
+  data <- data %>%
+    pivot_wider2(
+      names_from = .data$scenario,
+      values_from = c(
+        .data$weighted_production_target,
+        .data$weighted_technology_share_target
+      )
+    )
+
+  data <- data %>%
+    rename(
+      weighted_production_projected = .data$weighted_production,
+      weighted_technology_share_projected = .data$weighted_technology_share,
+      sector = .data$sector_ald
+    )
+
+  data %>%
+    pivot_longer(cols = starts_with("weighted_")) %>%
+    filter(!is.na(.data$value)) %>%
+    separate_metric_from_name() %>%
+    pivot_wider2()
+}
+
+separate_metric_from_name <- function(data) {
+  data %>%
+    mutate(
+      name = sub("weighted_", "", .data$name),
+      name = sub("(production)_", "\\1-", .data$name),
+      name = sub("(technology_share)_", "\\1-", .data$name)
```

Comment:
is "-" a proper "syntactic" character for data frame column name?

Reply:
This is only used intermittently, and is removed later in a pivot_longer (or wider, I can't remember).